src/main/java/org/ice4j/ice/Agent.java

```diff
@@ -1407,6 +1407,8 @@ public long getTieBreaker()
      */
     public void setControlling(boolean isControlling)
     {
+        this.logger.info(() -> "Changing agent " + this.toString() + " role from controlling = "
+            + this.isControlling + " to controlling = " + isControlling);
```

Comment:
It seems like this will log not only for role changes (which is good), but also on ordinary startup for a controlled node (which will be a lot of unnecessary noise).

Reply:
@JonathanLennox do you mean you want `if (this.isControlling != isControlling)` before printing log?

Reply:
No, I think what I mean is that ideally, the first time this is set, the "from" part of the log message wouldn't be included (since I feel it's not particularly meaningful in that case).

This would add state complexity which would only be used in log messages, though, so I'd be willing to let @bbaldino disagree with me.

Reply:
@JonathanLennox ok, I think I've got generic idea.
What do you think of updating this line from constructor:
https://github.com/jitsi/ice4j/blob/e854e4efaf5dbb82e45028fceafe9d76ca93cb2a/src/main/java/org/ice4j/ice/Agent.java#L405
To something like this:
```
logger.debug(() -> "Created a new Agent: " + this.toString() + " with ICE controlling role = " + this.isControling);
```
And in `setControlling` I could also add `if (this.isControlling != isControlling)` before printing. So it should solve the issue of unnecessary print to log when nothing is changed.

BTW, I could also update logging level to debug, so logs will only be polluted when detailed logs are enabled. What do you think?

Reply:
The problem with both of these is that `isControlling` isn't set in the constructor, it just defaults to `true`.  So for a controlled endpoint you'd always get "Created a new Agent: ... with ICE controlling role = true" and then "Changing Agent from controlling = true to controlling = false", every time.  This is what seems misleading to me.

Reply:
@JonathanLennox then the only way to resolve this issue seems to make initial ICE role as required constructor argument :) I could do this, but I will need to update JVB side to leverage new constructor (I'm actually working on PR for JVB which adjusts `Dtls/IceTransport` implementation, so it shouldn't be a problem at all to use potentially new `Agent`'s constructor) . If you think that it would be a good solution, I'll implement it, just let me know please. Thank you very much for review!

Reply:
i think the idea is sound, but given there doesn't seem to be a clean solution to get a better behavior, i think i'd rather leave it as-is than get into changing the ctor argument.  (i do, however, think that doing `if (this.isControlling != isControlling)` before printing log makes sense)